src/AdoNet/Shared/Storage/DbExtensions.cs

```diff
@@ -218,6 +232,19 @@ public static TValue GetValue<TValue>(this IDataRecord record, string fieldName)
             }
         }
 
```

Comment:
/// <summary>
        /// Returns a DateTime value with the given <see paramref="fieldName"/>.
        /// </summary>
        /// <param name="record">The record from which to retrieve the value.</param>
        /// <param name="fieldName">The name of the field.</param>
        /// <returns>DateTime Value in the given field.</returns>
        /// <exception cref="DataException"/>
        /// <remarks>An explicit function like this is needed in cases where to connector infers a type that is undesirable.
        /// An example here is Npgsql.NodaTime, which makes Npgsql return Noda type and consequently Orleans is not able to
        /// use it since it expects .NET DateTime. This function throws if the given <see paramref="fieldName"/> does not exist.</remarks>